src/shared/components/results/ResultsFacets.tsx

```diff
@@ -4,15 +4,29 @@ import { Facets, Facet, Loader } from 'franklin-sites';
 import useNS from '../../hooks/useNS';
 
 import TaxonomyFacet from './TaxonomyFacet';
+import EntryTypeIcon from '../entry/EntryTypeIcon';
 
 import { mainNamespaces } from '../../types/namespaces';
 
 import { UseDataAPIWithStaleState } from '../../hooks/useDataApiWithStale';
-import Response from '../../../uniprotkb/types/responseTypes';
+import Response, { FacetValue } from '../../../uniprotkb/types/responseTypes';
 
 import helper from '../../styles/helper.module.scss';
 import './styles/results-data.scss';
 
+const getDecoratedFacetLabel = (facetValue: FacetValue) => {
+  const { label } = facetValue;
+  if (typeof label === 'string') {
+    return (
+      <>
+        <EntryTypeIcon entryType={label} />
+        {facetValue.label}
+      </>
+    );
+  }
+  return facetValue.label;
```

Comment:
Not sure why destructuring the argument here but then using twice the destructured variable, and twice the original argument of the function.
I guess you should use one, or the other, but not a mixed approach.